docs/core/plugins/sieve.md

```diff
@@ -46,128 +58,191 @@ protocol lmtp {
-locations.
+Sieve scripts are retrieved from a script storage. This can currently be the
+local filesystem, an LDAP database or any dict storage. Depending on the storage
+implementation, its type and its configuration, storages can contain one script,
+several scripts identified by name, and a series of scripts in a well-defined
+order to be executed in sequence.
 
-The default [file](#file) location type is a directory containing
-one or more Sieve script files with a symlink pointing to the active
-one.
-
-More complex setups can use other location types such as [LDAP](#ldap) or
-[dict](#dict) to fetch Sieve scripts from remote databases.
-
-All settings that specify the location of one or more Sieve scripts
-accept the following syntax:
+Script storages are configured in a named sieve_script block:
 
 ```
-<setting> = [<type>:]path[;<option>[=<value>][;...]]
+sieve_script personal {
+  path = ~/.dovecot.sieve
+}
 ```
 
-If `<type>` is omitted, the script location type is `file` and the location
-is interpreted as a local filesystem path pointing to a Sieve script file
-or directory.
-
-### Common Settings
-
-All location types support the following settings:
-
-#### `name=<script-name>`
+The storage name (`personal` in the example) is used internally within
+configurations, as an identifier for logging, and as an identifier for command
+line tools. It also allows updating a storage that was defined earlier - by
+repeating the sieve_script block and adding additional configuration settings -
+or it allows userdb to override storage settings for specific users
+(FIXME: example?).
+
+### Script storage types
+
+Sieve scripts can be evaluated at various stages in message delivery and for
+stored messages. The type of the Sieve script storage determines where it is
+applicable, how the storage is accessed and how the retrieved Sieve script is
+evaluated.
 
-Set the name of the Sieve script that this location points to. If the
-name of the Sieve script is not contained in the location path and
-the location of a single script is specified, this option is required
-(e.g. for dict locations that must point to a particular script).
+The type of the Sieve script storage is configured using the
+[[setting,sieve_script_type]] setting. The following types are currently
+recognized (others are defined by the [[plugin,sieve-imapsieve]] plugin):
 
-If the name of the script is contained in the location path, the
-value of the name option overrides the name retrieved from the
-location. If the Sieve interpreter explicitly queries for a specific
-name (e.g. to let the Sieve [[link,sieve_include]] retrieve a script from
-the [[setting,sieve_global]]), this option has no effect.
+#### `personal`
 
-#### `bindir=<dirpath>`
+The `personal` storage serves as the user's main personal storage. Although more
+than a single `personal` storage can be defined, only the first one listed in
+the configuration is used. 
```

Comment:
Should it be an error instead?

Reply:
I am not deeply invested in this. Make a ticket if you want this changed.